python/lsst/daf/ingest/index_exposure_task.py

```diff
@@ -0,0 +1,358 @@
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+"""This module provides a |task| for spatial afw |exposure| indexing.
+
+:class:`.IndexExposureTask` extracts the WCS from an input exposure and uses
+it to compute a corresponding spherical bounding polygon. The exposure data-id
+and bounding polygon are then written to an SQLite 3 database.  Fast spatial
+queries are supported by maintaining an `R*Tree`_ index over exposures.
+
+.. _`R*Tree`:      https://www.sqlite.org/rtree.html
+.. |configuration| replace::  :class:`configuration <IndexExposureConfig>`
+.. |exposure|      replace::  :class:`exposure <lsst.afw.image.ExposureF>`
+.. |metadata|      replace::  :class:`metadata <lsst.daf.base.PropertySet>`
+.. |run|           replace::  :meth:`~IndexExposureTask.run`
+.. |runner|        replace::  :class:`runner <IndexExposureRunner>`
+.. |task|          replace::  :class:`~lsst.pipe.base.Task`
+"""
+
+try:
+    import cPickle as pickle
+except:
+    import pickle
+import sqlite3
+
+import lsst.daf.base as daf_base
+import lsst.afw.image as afw_image
+import lsst.pex.config as pex_config
+import lsst.pipe.base as pipe_base
+from lsst.sphgeom import ConvexPolygon, LonLat, UnitVector3d
+
+
+def quote_sqlite3_identifier(s):
+    """Safely quote a string `s` for use as an SQLite 3 identifier.
+
+    Note that `s` is required to be valid UTF-8 (or encodable as such), and
+    may not contain embedded NUL characters.
+
+    This function exists because Python DB-API parameter substitution does not
+    work for table and column names. Without proper quoting, they present
+    an opportunity for SQL injection.
+    """
+    if isinstance(s, unicode):
+        # Convert to a UTF-8 string
+        ident = s.encode('utf-8')
+    else:
+        # Make sure s is valid UTF-8
+        ident = s.decode('utf-8').encode('utf-8')
+    if ident.find('\x00') >= 0:
+        raise RuntimeError('The NUL character is not legal '
+                           'in SQLite 3 identifiers')
+    # Quote the identifier. Embedded quotes are escaped by doubling them up.
+    return '"' + ident.replace('"', '""') + '"'
+
+
+class IndexExposureConfig(pex_config.Config):
+    """Configuration for :class:`~IndexExposureTask`."""
+
+    allow_replace = pex_config.Field(
```

Comment:
I think I'm going to stick with allow_replace - replace is databasey terminology for clobber, and it's consistent with the catalog ingestion task.